rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -515,7 +515,23 @@ bool main_window::InstallRapFile(const QString& path, const std::string& filenam
 	{
 		return false;
 	}
-	return fs::copy_file(sstr(path), Emulator::GetHddDir() + "/home/" + Emu.GetUsr() + "/exdata/" + filename, true);
+
```

Comment:
This is way too complicated for a simple copy. Either write your own copy function or improve the existing one. But adding your overdesigned code everywhere is not gonna make anyone happy.

Reply:
Let's say I rewrite fs::copy_file to be atomic, there is still functionality missing here for error checking (< 0x10 check, which must happen with the same open file handle). Seems odd to pass a function for fs::copy_file to do error checking. IMO this **unique* one use can remain in this form until there will be more like it.

Reply:
I added comments though.

Reply:
if it's only the file size you want to know, then you can just check for info.size() in IsValidFile

Reply:
What if another thread or process changes its size inbetween?

Reply:
this is just paranoid. I don't see why this would ever happen and why we should care.

Reply:
whatever. just ad the info.size() check and test if it works. I don't really care if there's more code in this function.
I just think that it's not really necessary to implement such a fortress for such a trivial task.

Reply:
Done.